packages/@react-spectrum/menu/src/Menu.tsx

```diff
@@ -55,6 +56,15 @@ function Menu<T extends object>(props: SpectrumMenuProps<T>, ref: DOMRef<HTMLDiv
     }
   }, []);
 
+  useInteractOutside({
+    ref: domRef,
+    onInteractOutside: (e) => {
+      if (!isSubmenu && !popoverContainerRef.current?.contains(e.target) && !trayContainerRef.current?.contains(e.target)) {
```

Comment:
also if the target isn't contained within the root menu?

Reply:
Don't think we need to check for that since onInteractOutside will already only trigger for events happening outside the root menu

Reply:
o, good call, you are correct